### PR TITLE
feat: Table supports hidden columns

### PR DESCRIPTION
**Pull request by madocto**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close: #46930 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Table supports hidden columns      |
| 🇨🇳 Chinese |    Table 组件支持隐藏列       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input/style/token.ts [hunk 1]
```diff
@@ -145,8 +146,8 @@ export const initComponentToken = (token: AliasToken): SharedComponentToken => {
     activeShadow: `0 0 0 ${controlOutlineWidth}px ${controlOutline}`,
     errorActiveShadow: `0 0 0 ${controlOutlineWidth}px ${colorErrorOutline}`,
     warningActiveShadow: `0 0 0 ${controlOutlineWidth}px ${colorWarningOutline}`,
-    hoverBg: '',
-    activeBg: '',
+    hoverBg: colorBgContainer,
+    activeBg: colorBgContainer,
```
Comment (afc163) [inline components/input/style/token.ts:150]: 不应该改这个。

**Comment by afc163** [comment]
混入了其他 commit。

File: components/table/index.zh-CN.md [hunk 2]
```diff
@@ -209,6 +210,7 @@ const columns = [
 | sortIcon | 自定义 sort 图标 | (props: { sortOrder }) => ReactNode | - | 5.6.0 |
 | title | 列头显示文字（函数用法 `3.10.0` 后支持） | ReactNode \| ({ sortOrder, sortColumn, filters }) => ReactNode | - |  |
 | width | 列宽度（[指定了也不生效？](https://github.com/ant-design/ant-design/issues/13825#issuecomment-449889241)） | string \| number | - |  |
+| hidden | 隐藏列 | boolean | false | 5.12.9 |
```
Comment (afc163) [inline components/table/index.zh-CN.md:213]:
```suggestion
| hidden | 隐藏列 | boolean | false | 5.13.0 |
```

File: components/table/demo/hidden-columns.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    key: '2',
+    name: 'Jim Green',
+    age: 40,
+    address: 'London Park',
+  },
+];
+
+const App: React.FC = () => <Table columns={columns} dataSource={data} scroll={{ x: 1300 }} />;
+
+export default App;
```
Comment (afc163) [inline components/table/demo/hidden-columns.tsx:62]: 做一个 Checkbox Group 点击联动把，这样更容易看出 hidden 的作用。否则这个例子外表看上去就是个普通表格。

File: components/table/demo/hidden-columns.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  {
+    key: '2',
+    name: 'Jim Green',
+    age: 40,
+    address: 'London Park',
+  },
+];
+
+const App: React.FC = () => <Table columns={columns} dataSource={data} scroll={{ x: 1300 }} />;
```
Comment (afc163) [inline components/table/demo/hidden-columns.tsx:60]: 貌似没必加 `scroll` 属性。

**Comment by xrkffgg** [comment]
@madocto 老哥，要不你重新开个 PR 往 master 合吧

**Comment by madocto** [comment]
抱歉，没明白怎么混入了其它 commit，我重新开个PR

**Comment by yoyo837** [comment]
是分支问题不是PR问题，分支rebase一下

**Pull request closed without merging**